lib/taskana-core/src/main/java/pro/taskana/TaskanaEngine.java

```diff
@@ -52,6 +55,12 @@
      */
     TaskanaEngineConfiguration getConfiguration();
 
+    /**
+     * The HistoryEventProducer events and emits them to the registered history service providers.
+     * @return The HistoryEventProducer
+     */
+    HistoryEventProducer getHistoryEventProducer();
```

Comment:
This is supposed to be an internal operation. Please move it to the Internal interface